elastictransport/elastictransport.go

```diff
@@ -276,11 +279,19 @@ func (c *Client) Perform(req *http.Request) (*http.Response, error) {
 	if req.Body != nil && req.Body != http.NoBody {
 		if c.compressRequestBody {
 			var buf bytes.Buffer
-			zw := gzip.NewWriter(&buf)
-			if _, err := io.Copy(zw, req.Body); err != nil {
+			level := gzip.DefaultCompression
```

Comment:
The default value should be initialized in the `New` function, relevant field in `Client` are not exposed and will be set by the constructor either way.

Reply:
Thanks, amended.